src/hooks/useKeyboardNavigation.ts

```diff
@@ -0,0 +1,313 @@
+import { useEffect } from 'react';
+import tinykeys from 'tinykeys';
+
+import {
+  focusElement,
+  focusNextCategory,
+  focusPrevCategory,
+  getActiveElement,
+  getCurrentEmojiListGroup,
+  getElementBoundariesInfo,
+  getEmojiGroupName,
+  getGridInfo,
+  getNextEmoji,
+  getPrevEmoji,
+  withCatch
+} from '../DomUtils/keyboardNavigation';
+
+import {
+  useActiveCategoryState
+  // useActiveSkinTone,
+  // useSetActiveCategory,
+  // useSkinToneSpreadValue,
+  // useToggleSpreadSkinTones
+} from '../components/contextProvider/PickerContextProvider';
+import { DOWN, LEFT, RIGHT, UP } from './consts';
+
+const useKeyboardNavigation = ({
+  categoriesNavRef,
+  emojiSearchRef,
+  emojiListRef,
+  skinToneSpreadRef
+}: any) => {
+  const [, setCategory] = useActiveCategoryState();
+  //const setActiveCategory = useSetActiveCategory();
+  // const toggleSkinTonesSpread = useToggleSpreadSkinTones();
+  // const activeSkinTone = useActiveSkinTone();
+  // const isSkinToneSpreadOpen = useSkinToneSpreadValue();
+
+  useEffect(() => {
+    return tryTinyKeys(categoriesNavRef.current, {
+      ArrowLeft: focusPrevCategory,
+      ArrowRight: focusNextCategory,
+      ArrowDown: focusNextSection,
+      Escape: exitEmojiList
+    });
+  }, []);
+
+  useEffect(() => {
+    if (!emojiSearchRef.current) return;
+
+    return tryTinyKeys(emojiSearchRef.current, {
+      ArrowRight: focusSkinTonePicker,
+      ArrowUp: focusPrevSection,
+      ArrowDown: focusNextSection
+    });
+  }, []);
+
+  useEffect(() => {
+    if (!emojiSearchRef.current) return;
+
+    // return tryTinyKeys(skinToneSpreadRef.current, {
+    //   // ArrowLeft: isSkinToneSpreadOpen ? focusNextSkinTone : exitSkinTones,
+    //   ArrowRight: focusPrevSkinTone,
+    //   Escape: exitSkinTones,
+    //   Enter: exitSkinTones
+    // });
+  }, []);
+
+  useEffect(() => {
+    return tryTinyKeys(emojiListRef.current, {
+      ArrowRight: withPreventDefault(() => navigateGrid(RIGHT)),
+      ArrowLeft: withPreventDefault(() => navigateGrid(LEFT)),
+      ArrowUp: withPreventDefault(() => navigateGrid(UP)),
+      ArrowDown: withPreventDefault(() => navigateGrid(DOWN)),
+      Escape: exitEmojiList
+    });
+  }, []);
+
+  function exitEmojiList() {
+    focusSearch();
+    scrollEmojiListToTop();
+  }
+
+  const updateActiveItem = (newActiveItem: any) => {
+    focusElement(newActiveItem);
+  };
+
+  const navigateGrid = withCatch((direction: any) => {
```

Comment:
now instead of `any`, you can do 
```suggestion
  const navigateGrid = withCatch((direction: Directions) => {
```